java-frontend/src/main/java/org/sonar/java/resolve/TypeSubstitution.java

```diff
@@ -76,4 +76,20 @@ public int hashCode() {
     return this.substitutionEntries().hashCode();
   }
 
+  public boolean isIdentity() {
+    for (Map.Entry<TypeVariableJavaType, JavaType> substitution : this.substitutionEntries()) {
+      if (substitution.getKey() != substitution.getValue()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)